src/index.js

```diff
@@ -22,35 +38,37 @@ function decode(res) {
   return JSON.parse(jsonPayload);
 }
 
-function backwardCompat(node, attr, key) {
-  return `${node}[data-mystique-key="${key}"],${node}[data-${attr}="${key}"]`;
+function toScript(attr, key, props, id) {
+  return `<script type="application/json" data-${attr}="${key}" data-hypernova-id=${id}>${LEFT}${encode(props)}${RIGHT}</script>`; // eslint-disable-line max-len
 }
 
-function toScript(attr, key, props) {
-  return `<script type="application/json" data-${attr}="${key}">${LEFT}${encode(props)}${RIGHT}</script>`; // eslint-disable-line max-len
-}
-
-function fromScript(attr, key) {
-  const node = document.querySelector(backwardCompat('script', attr, key));
+function fromScript(attr, key, id) {
+  const idSelector = id ? `[data-hypernova-id="${id}"]` : '';
```

Comment:
No, I'm ok having it blow up if people pass the wrong input. This is developer input not user input so it's ok.